### ISSUE TITLE
vouchers with product selection not applied correctly if multiple products are in shopping cart

### ISSUE THREAD

**Issue by boxledev** [PrestaShop/PrestaShop#22754]
When choosing product selection for vouchers, the vouchers are also apllied to not eligible products if there are also eligible products in the shopping cart.
this is correct behaviour when only a product that is not eligible for the voucher is in the shopping cart:
![correct](https://user-images.githubusercontent.com/48062599/104093309-17037880-528a-11eb-876e-a4a95ac5fb76.PNG)
now when adding any eligible product, the voucher is also applied to to the wrong product (in this case it is a 10% discount, so should only be 0.30€ discount).
![2products](https://user-images.githubusercontent.com/48062599/104093380-84afa480-528a-11eb-8699-e88997a48e68.PNG)

* PrestaShop version: 1.7.7.0
* PHP version: 7.2

**Comment by hibatallahAouadni** [comment]
Hello @boxledev 

Please **follow the template to submit a complete bug report** and fill the blocks that we provided in the template (describe the bug, expected behavior, steps to reproduce, ..).

Your issue is kind of similar to this one #22057
Could you please check and feedback.

Thanks!

**Comment by boxledev** [comment]
I'm terribly sorry, in trying to reproduce the issue I noticed I had enabled the wrong setting in the Apply discount to settting under the actions tab. It was set to order instead of selected products. It now uses the correct discount rules. A bit counterintuitive to select products for a voucher and to have to go to the other tab to actually take that into account, but the mistake was mine. 
Apologies again
